google-cloud-firestore/src/main/java/com/google/cloud/firestore/BulkWriter.java

```diff
@@ -989,7 +989,9 @@ private void sendBatchLocked(final BulkCommitBatch batch, final boolean flush) {
                 @Override
                 public void run() {
                   synchronized (lock) {
-                    scheduleCurrentBatchLocked(flush);
+                    if (flush) {
```

Comment:
The if-check can happen outside the synchronized block.

Reply:
done.